apis/nucleus/src/components/listbox/components/ListBoxRowColumn/ListBoxRowColumn.jsx

```diff
@@ -91,32 +92,14 @@ function RowColumn({ index, rowIndex, columnIndex, style, data }) {
 
   const handleKeyDownCallback = useCallback(getFieldKeyboardNavigation({ ...actions, focusListItems }), [actions]);
 
-  const [isSelected, setSelected] = useState(false);
-  const [cell, setCell] = useState();
-
+  const cell = useMemo(() => getCellFromPages({ pages, cellIndex }), [pages, cellIndex]);
   const [classArr, setClassArr] = useState([]);
-
-  useEffect(() => {
-    if (!pages) {
-      return;
-    }
-    let c;
-    const page = pages.filter((p) => p.qArea.qTop <= cellIndex && cellIndex < p.qArea.qTop + p.qArea.qHeight)[0];
-    if (page) {
-      const area = page.qArea;
-      if (cellIndex >= area.qTop && cellIndex < area.qTop + area.qHeight) {
-        [c] = page.qMatrix[cellIndex - area.qTop];
-      }
-    }
-    setCell(c);
-  }, [pages]);
+  const isSelected = cell?.qState === 'S' || cell?.qState === 'XS' || cell?.qState === 'L';
 
   useEffect(() => {
```

Comment:
also changing this use useMemo
(`const classArr = useMemo(() => ..., [cell?.qState, histogram, dense, checkboxes])`)
will probably save one more render (from removing the setClassArr call)

Reply:
Yes that works as well 👍